api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java

```diff
@@ -9913,6 +9990,83 @@ public CohortDefinition getPatientsWhoAbandonedTarvOnArtStartDateForPregnants()
+            + "                                                                 AND e.encounter_type = ${6}  "
+            + "                                                                 AND o.concept_id = ${6273}  "
+            + "                                                                 AND o.value_coded = ${1707}  "
+            + "                                                                 AND e.location_id = :location  "
+            + "                                                                 AND e.encounter_datetime > DATE_SUB(end_period.first_gestante, INTERVAL 3 MONTH)  "
+            + "                                                                 AND e.encounter_datetime <= end_period.first_gestante  "
+            + "                                                               GROUP BY p.patient_id  "
+            + "                                                               UNION  "
+            + "                                                               SELECT p.patient_id, max(o.obs_datetime) as last_encounter FROM patient p  "
+            + "                                                                                                                                   INNER JOIN encounter e ON e.patient_id = p.patient_id  "
+            + "                                                                                                                                   INNER JOIN obs o on e.encounter_id = o.encounter_id  "
+            + "                                                                                                                                   INNER JOIN (  "
+            + QualityImprovement2020Queries.getPregnancyDuringPeriod()
+            + "                                                               ) end_period ON end_period.patient_id = p.patient_id  "
+            + "                                                               WHERE p.voided = 0 AND e.voided = 0 AND o.voided = 0  "
+            + "                                                                 AND e.encounter_type = ${53}  "
+            + "                                                                 AND o.concept_id = ${6272}  "
+            + "                                                                 AND o.value_coded = ${1707}  "
+            + "                                                                 AND e.location_id = :location  "
+            + "                                                                 AND e.encounter_datetime > DATE_SUB(end_period.first_gestante, INTERVAL 3 MONTH)  "
+            + "                                                                 AND e.encounter_datetime <= end_period.first_gestante  "
+            + "                                                               GROUP BY p.patient_id  "
+            + "                                                           ) abandoned GROUP BY abandoned.patient_id";
+
```

Comment:
The query is not implementing what is being told on the description, pls update the javadocs.
The description talks about 3 months after the ART startDate and the query implements last 3 months before pregnancy date

Reply:
Solved